sale_rental/models/sale_order.py

```diff
@@ -0,0 +1,286 @@
+    rental_type = fields.Selection([
+        ('new_rental', 'New Rental'),
+        ('rental_extension', 'Rental Extension')], 'Rental Type',
+        readonly=True, states={'draft': [('readonly', False)]})
+    extension_rental_id = fields.Many2one(
+        'sale.rental', string='Rental to Extend')
+    rental_qty = fields.Float(
+        string='Rental Quantity',
+        digits=dp.get_precision('Product Unit of Measure'),
+        help="Indicate the number of items that will be rented.")
+    sell_rental_id = fields.Many2one(
+        'sale.rental', string='Rental to Sell')
+
+    _sql_constraints = [(
+        'rental_qty_positive',
+        'CHECK(rental_qty >= 0)',
+        'The rental quantity must be positive or null.'
+        )]
+
+    @api.constrains(
+        'rental_type', 'extension_rental_id', 'start_date', 'end_date',
+        'rental_qty', 'product_uom_qty', 'product_id')
+    def _check_sale_line_rental(self):
+        for line in self:
+            if line.rental_type == 'rental_extension':
+                if not line.extension_rental_id:
+                    raise ValidationError(_(
+                        "Missing 'Rental to Extend' on the sale order line "
+                        "with rental service %s") % line.product_id.name)
+
+                if line.rental_qty != line.extension_rental_id.rental_qty:
+                    raise ValidationError(_(
+                        "On the sale order line with rental service %s, "
+                        "you are trying to extend a rental with a rental "
+                        "quantity (%s) that is different from the quantity "
+                        "of the original rental (%s). This is not supported.")
+                        % (
+                        line.product_id.name,
+                        line.rental_qty,
+                        line.extension_rental_id.rental_qty))
+            if line.rental_type in ('new_rental', 'rental_extension'):
+                if not line.product_id.rented_product_id:
+                    raise ValidationError(_(
+                        "On the 'new rental' sale order line with product "
+                        "'%s', we should have a rental service product !") % (
+                        line.product_id.name))
+                if line.product_uom_qty !=\
+                        line.rental_qty * line.number_of_days:
+                    raise ValidationError(_(
+                        "On the sale order line with product '%s' "
+                        "the Product Quantity (%s) should be the "
+                        "number of days (%s) "
+                        "multiplied by the Rental Quantity (%s).") % (
+                        line.product_id.name, line.product_uom_qty,
+                        line.number_of_days, line.rental_qty))
+                # the module sale_start_end_dates checks that, when we have
+                # must_have_dates, we have start + end dates
+            elif line.sell_rental_id:
```

Comment:
```suggestion
            elif line.sell_rental_id and line.product_uom_qty != line.sell_rental_id.rental_qty:

```

Reply:
line length issue. I prefer keeping the original code as is.